### DISCUSSION TITLE
[Question] Add AuthGuards to create or update in auto generated resolver

### DISCUSSION THREAD

**Discussion by ThianHooi** [doug-martin/nestjs-query#1502] [Q&A]
I would like to add auth guards to certain graphql mutation, e.g. create or update in auto generated resolver. Please provide an example. Any help is greatly appreciated!

I found this section from the [docs](https://doug-martin.github.io/nestjs-query/docs/graphql/authorization#authorize-depending-on-operation).

What I did are
1. Create a custom authorizer which resolve when the operation is `readonly: true` 
2. Remove the `guards: [JwtAuthGuard]` in the module (not sure if this is right)
3. Add the `@Authorize(CustomAuthorizer)` on dto

But this is for authorization, which is deciding who can access what data. I am trying to achieve authentication in selected operations or query.

**Comment by smolinari** [comment]
@ThianHooi - What nestjs-query is offering are points to add authorization, not authentication. Authentication should be global to your app, theoretically. By that I mean, if you have users being authenticated, the whole app needs to be covered by your authentication guard (global) and then, depending on how the user authenticated (i.e. logged in or not) you allow access (which is then authorization). 

Scott
